spec/gogo/x/username/codec.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package username;
+
+import "codec.proto";
+import "gogoproto/gogo.proto";
+
+// Token model represents a username mapping to an address together with all
+// metadata.
+//
+// Each Token model is stored using the username as the key. This guarantee
+// that the name is unique. Username is a combination of a name and a domain.
+// The format is <name>*<domain>
+//
+// Each token points to a blockchain and an address on that blockchain. Both
+// blockchain ID and address are an arbitrary string as we do not want to limit
+// ourselves to certain patterns.
+message Token {
+  weave.Metadata metadata = 1;
+  // Target specifies where this username token points to.
+  Location target = 2 [(gogoproto.nullable) = false];
+  // Owner is a weave.Address that controls this token. Only the owner can
+  // modify a username token.
+  bytes owner = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+}
+
+// Location represents a blochain address. This structure clubs together
+// blokchain ID together with an address on that network. It is used to point
+// to an address on any blockchain network.
+message Location {
```

Comment:
pp: naming.... I found this more confusing than `ChainAddress`

Reply:
I second that - looks confusing to me, as I start thinking about physical location of things. 
I would either keep `ChainAddress` which is not fantastic, or introduce an `Address` entity, which makes sense in terms of NFT. In the end `ChainAddress` is more self-explanatory. 

It could also be `ExternalAddress` or something similar.

Reply:
Done in https://github.com/iov-one/weave/pull/771/commits/de3d772fd6c3508e8db76d2d169399f583a431f7 I have renamed it to `BlockchainAddress`.